### PR TITLE
DM-33589: Compatibility updates for upstream resync

### PR DESCRIPTION
**Pull request by mfisherlevine**

### CODE REVIEW

File: python/lsst/atmospec/formatters.py [hunk 1]
```diff
@@ -31,10 +33,27 @@ class SpectractorSpectrumFormatter(FileFormatter):
     unsupportedParameters = None
 
     def _readFile(self, path, pytype=None):
-        return Spectrum(path)
+        spectrogramPath = path + 'spectrogram.fits'
```
Comment (czwa) [inline python/lsst/atmospec/formatters.py:36]: Does this yield file names like `/path/to/spectrum.ext`, `/path/to/spectrum.extspectrogram.fits`, and `/path/to/spectrum.exttable.csv`?

File: python/lsst/atmospec/formatters.py [hunk 2]
```diff
@@ -31,10 +33,27 @@ class SpectractorSpectrumFormatter(FileFormatter):
...
+        # Spectractor supports that
+        if os.path.isfile(spectrogramPath) and os.path.isfile(psfPath):
+            return Spectrum(path, fast_load=False,
+                            spectrogram_file_name_override=spectrogramPath,
+                            psf_file_name_override=psfPath)
+        else:
+            return Spectrum(path, fast_load=True)
 
     def _writeFile(self, inMemoryDataset):
+        spectrogramFilename = self.fileDescriptor.location.path + 'spectrogram.fits'
```
Comment (czwa) [inline python/lsst/atmospec/formatters.py:48]: Same question here.

File: python/lsst/atmospec/processStar.py [hunk 3]
```diff
@@ -91,6 +91,12 @@ class ProcessStarTaskConnections(pipeBase.PipelineTaskConnections,
         storageClass="SpectractorImage",
         dimensions=("instrument", "visit", "detector"),
     )
+    spectractorPickle = cT.Output(
```
Comment (czwa) [inline python/lsst/atmospec/processStar.py:94]: See comment on DM-33326 review, re: just calling this a "spectraction".

File: python/lsst/atmospec/spectraction.py [hunk 4]
```diff
@@ -205,7 +207,7 @@ def _setImageAndHeaderInfo(self, image, exp, useVisitInfo=True):
                 vi = exp.getInfo().getVisitInfo()
                 image.header.airmass = vi.getBoresightAirmass()  # currently returns nan for obs_ctio0m9
                 image.airmass = vi.getBoresightAirmass()  # currently returns nan for obs_ctio0m9
-
+                image.date_obs = vi.date.toString(DateTime.UTC)  # should this be UTC or TAI?
```
Comment (czwa) [inline python/lsst/atmospec/spectraction.py:210]: Do you have a ticket that will answer this question?

File: python/lsst/atmospec/spectraction.py [hunk 5]
```diff
@@ -405,44 +414,10 @@ def run(self, exp, xpos, ypos, target, outputRoot=None, plotting=True):
         # Full forward model extraction:
         # adds transverse ADR and order 2 subtraction
         w = None
-        if parameters.PSF_EXTRACTION_MODE == "PSF_2D" and parameters.OBS_OBJECT_TYPE == "STAR":
-            w = FullForwardModelFitWorkspace(spectrum, verbose=1, plot=True, live_fit=False,
+        if parameters.SPECTRACTOR_DECONVOLUTION_FFM:
```
Comment (czwa) [inline python/lsst/atmospec/spectraction.py:417]: Should this be mutually exclusive with the `parameters.SPECTRACTOR_DECONVOLUTION_PSF2D`?  As is, it looks like you could potentially attempt to deconvolve this twice.

**Review by czwa — APPROVED** [review]
Generally just questions to make sure I understand what's going on.  This also needs to match any changes on DM-33326.

File: python/lsst/atmospec/formatters.py [hunk 6]
```diff
@@ -31,10 +33,27 @@ class SpectractorSpectrumFormatter(FileFormatter):
...
+        linesFilename = self.fileDescriptor.location.path + 'lines.csv'
+        psfFilename = self.fileDescriptor.location.path + 'table.csv'
```
Comment (timj) [inline python/lsst/atmospec/formatters.py:50]: Um. Hang on. This code fundamentally breaks the way that butler formatters work. It's never going to work with an object store. A formatter can only write one file or else it is a disassembled composite storage class where butler knows that multiple files are going to be written out and where they are. @mfisherlevine we are going to have to reorganize all of this and think deeply.

**Review by timj — CHANGES_REQUESTED** [review]
I will formally mark this as request changes because it needs a complete rethink.

File: python/lsst/atmospec/spectraction.py (hunk 5 above)
Comment (mfisherlevine) [reply python/lsst/atmospec/spectraction.py:418]: Agreed. I can raise that with Jérémy, but this top level file is designed to just follow the top-level workflow from upstream, and I've checked and that's how things are there. Will see if he just treats them as mutually exclusive though, or if they are ever run together or what.

**Pull request closed without merging**